backend/models/bcf.js

```diff
@@ -408,816 +464,639 @@ function getBCFMarkup(issue, account, model, unit) {
+										const groupDbCol = {
+											account: account,
+											model: model
+										};
 
-						if (_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0]")) {
-							vp.up = [
-								parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].CameraUpVector[0].X[0]._")),
-								parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].CameraUpVector[0].Z[0]._")),
-								-parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].CameraUpVector[0].Y[0]._"))
-							];
-							vp.view_dir = [
-								parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].CameraDirection[0].X[0]._")),
-								parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].CameraDirection[0].Z[0]._")),
-								-parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].CameraDirection[0].Y[0]._"))
-							];
-							vp.position = [
-								parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].CameraViewPoint[0].X[0]._")) * scale,
-								parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].CameraViewPoint[0].Z[0]._")) * scale,
-								-parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].CameraViewPoint[0].Y[0]._")) * scale
-							];
-
-							vp.fov = parseFloat(_.get(vpXML, "VisualizationInfo.PerspectiveCamera[0].FieldOfView[0]._")) * Math.PI / 180;
-
-							vp.type = "perspective";
-
-						} else if (_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0]")) {
-
-							vp.up = [
-								parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraUpVector[0].X[0]._")),
-								parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraUpVector[0].Z[0]._")),
-								-parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraUpVector[0].Y[0]._"))
-							];
-
-							vp.view_dir = [
-								parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraDirection[0].X[0]._")),
-								parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraDirection[0].Z[0]._")),
-								-parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraDirection[0].Y[0]._"))
-							];
-
-							vp.position = [
-								parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraViewPoint[0].X[0]._")) * scale,
-								parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraViewPoint[0].Z[0]._")) * scale,
-								-parseFloat(_.get(vpXML, "VisualizationInfo.OrthogonalCamera[0].CameraViewPoint[0].Y[0]._")) * scale
-							];
-
-							vp.fov = 1.8;
-
-							vp.type = "orthogonal";
-						}
+										const vpComponents = _.get(vpXML, "VisualizationInfo.Components");
 
-						if (_.get(vpXML, "VisualizationInfo.Components")) {
-							const groupDbCol = {
-								account: account,
-								model: model
-							};
+										for (let i = 0; i < vpComponents.length; i++) {
 
-							const vpComponents = _.get(vpXML, "VisualizationInfo.Components");
+											let highlightedGroupObject;
```

Comment:
I think we should do const here and setup the basic group stuff instead of calling createGroupObject and doing all these if checks everytime. i.e.

```suggestion
const highlightedGroupObject = {issue,name, color: [255,0,0], objects: {}};
```

then within those loops, all you have to do is
````
group.objects[model] =  group.objects[model] || [];
group.objects[model].push(id)
````

Reply:
Or, since that highlghtedGroupObjects isn't exactly in group schema anyway (you still have to call createGroupData after....), just leave the fluff out till you're about to create the group, ie only store what you care about:
````
{
    "<model id>": [ ifc guids]
}
````